Codec/Archive/Tar/Types.hs

```diff
@@ -342,7 +342,7 @@ fromTarPathToWindowsPath (TarPath namebs prefixbs) = adjustDirectory $
   where
     name   = BS.Char8.unpack namebs
     prefix = BS.Char8.unpack prefixbs
-    adjustDirectory | BS.Char8.pack [FilePath.Posix.pathSeparator] `BS.isPrefixOf` namebs
+    adjustDirectory | BS.Char8.singleton FilePath.Posix.pathSeparator `BS.isPrefixOf` namebs
```

Comment:
You fixed `isPrefixOf` above, but not here. 

I would use `BS.last` instead of `BS.isSuffixOf`.

Reply:
Is it also necessary to check that `namebs` is non-empty?

Reply:
Yes.

Reply:
That was why I used `isPrefixOf`. I'll change it.